src/Microsoft.OData.Core/UriParser/Binders/SelectExpandBinder.cs

```diff
@@ -984,9 +986,11 @@ private static void AddToSelectedItems(SelectItem itemToAdd, List<SelectItem> se
             }
 
             // if the selected item is "*", filter the existing property selection.
-            if (itemToAdd is WildcardSelectItem)
+            wildcardSelectItem = itemToAdd as WildcardSelectItem;
+            if (wildcardSelectItem != null)

```

Comment:
```suggestion
            if (itemToAdd is WildcardSelectItem wildcardSelectItem)
```

Reply:
We're reusing `wildcardSelectItem` from above so cannot redefine it here.